### ISSUE TITLE
Resolution reverts after exiting game

### ISSUE THREAD

**Issue by GimmickyMoniker** [Nonary/ResolutionAutomation#76]
i'm streaming to a secondary monitor that's 1080p (my main display is 2K), when I start moonlight and go to desktop it successfully matches resolution, however, with most games I play when I exit the game, my resolution goes back to 2K despite playing on the 1080p monitor. Please advise how I can fix this.

**Comment by Nonary** [comment]
The script is designed for one monitor only, please use https://github.com/Nonary/MonitorSwapAutomation to force your machine to swap to one monitor while streaming and that should resolve your issue

**Comment by GimmickyMoniker** [comment]
I'm confused as to how Moonlight can be only ran with one monitor, isn't the whole point of the program to stream games/programs ect from a PC to a second screen? are there people running it with no monitor at all? The monitor attached to my PC is not on while I run moonlight.

**Comment by GimmickyMoniker** [comment]
When i say secondary monitor, I mean I'm streaming moonlight through a Steam Link hooked up HDMI to another display, I'm not running multiple monitors hooked up to my computer at the same time.

**Comment by Nonary** [comment]
**Check the Log File**: The script log file can help determine if the resolution change is being properly handled. Please upload the log.txt file located in the folder where the script is installed (or in the logs folder, sorted by date). This will allow me to see if the script successfully changes the resolution after exiting the game.

**Borderless Windowed Mode**: Ensure that your games are running in borderless windowed mode instead of fullscreen. The script works best when games are set to borderless windowed mode because it inherits the resolution from the desktop games that use fullscreen modes will not work with the script.

**Comment by GimmickyMoniker** [comment]
[log.txt](https://github.com/user-attachments/files/16680888/log.txt)

There are several games I play that don't support borderless window mode. It's frustrating it resets to 2K when leaving a game, even if the game's resolution is 1080p. this didn't happen when I used the version of this script that supported GFE.

**Comment by Nonary** [comment]
> [log.txt](https://github.com/user-attachments/files/16680888/log.txt)
> 
> There are several games I play that don't support borderless window mode. It's frustrating it resets to 2K when leaving a game, even if the game's resolution is 1080p. this didn't happen when I used the version of this script that supported GFE.

That logs shows that it has been set to 1080p but did not make any changes afterwards, so if you're noticing that the desktop keeps changing to a specific resolution it is unrelated to this script. Are you running any other scripts with Sunshine?

**Comment by GimmickyMoniker** [comment]
This is the only one I've downloaded. The only changes I've made is I used the GFE version then switched to the "main" one after updating my Nviida App and now it does this.

**Comment by GimmickyMoniker** [comment]
It seems something about the launching of the games this happens to sets the resolution back to the default

**Comment by xDaveN** [comment]
@Nonary The title of this issue represents what I'm experiencing, but my setup is different.
I'm streaming from a 3440x1440 display to a 4K TV through a NVIDIA Shield. My moonlight is setup to receive 1080p 120fps.
I launch my stream into Playnite (a game launcher) which successfully prompts the resolution switch to 1080p. Launching a game retains the correct resolution, however whenever I close said game my resolution reverts to 3440x1440 - I would like to remain in 1080p within Playnite, in case I want to switch to another game.

Not sure if this issue is related to your script, or if Playnite somehow forces my monitors native res. Would appreciate some advice.
 :)

**Comment by KlosiakMK** [comment]
Same here and as above. I am using the Sunshine + Moonlight combo to stream games in 4K resolution. I have 3840x1600 monitor connected to my host PC and custom resolution of 3840x2160 120Hz added because this is the stream parameters that I am using for my 4K TV. I am streaming my host PC desktop and running all the games from the streamed desktop. When starting desktop stream everything works fine and 3840x1600 resolution is switched to 3840x2160. The problem is when game/app use fullscreen mode. After closing game/app resolution of a streamed desktop is reverted to 3840x1600 and I have black bars on the top/bottom of my TV. I can set 4K resolution manually and this still works but I am wondering why script is behaving in such a way. The issue can be easily recreated using Heaven/Valley benchmarks in fullscreen setting on a streamed desktop.

**Comment by Nonary** [comment]
> Same here and as above. I am using the Sunshine + Moonlight combo to stream games in 4K resolution. I have 3840x1600 monitor connected to my host PC and custom resolution of 3840x2160 120Hz added because this is the stream parameters that I am using for my 4K TV. I am streaming my host PC desktop and running all the games from the streamed desktop. When starting desktop stream everything works fine and 3840x1600 resolution is switched to 3840x2160. The problem is when game/app use fullscreen mode. After closing game/app resolution of a streamed desktop is reverted to 3840x1600 and I have black bars on the top/bottom of my TV. I can set 4K resolution manually and this still works but I am wondering why script is behaving in such a way. The issue can be easily recreated using Heaven/Valley benchmarks in fullscreen setting on a streamed desktop.

Without the logs I can't really troubleshoot your issue please upload them

**Comment by KlosiakMK** [comment]
> > Same here and as above. I am using the Sunshine + Moonlight combo to stream games in 4K resolution. I have 3840x1600 monitor connected to my host PC and custom resolution of 3840x2160 120Hz added because this is the stream parameters that I am using for my 4K TV. I am streaming my host PC desktop and running all the games from the streamed desktop. When starting desktop stream everything works fine and 3840x1600 resolution is switched to 3840x2160. The problem is when game/app use fullscreen mode. After closing game/app resolution of a streamed desktop is reverted to 3840x1600 and I have black bars on the top/bottom of my TV. I can set 4K resolution manually and this still works but I am wondering why script is behaving in such a way. The issue can be easily recreated using Heaven/Valley benchmarks in fullscreen setting on a streamed desktop.
> 
> Without the logs I can't really troubleshoot your issue please upload them

Hi Nonary. Sure thing about the logs. Which one are needed and how to gdy then?

**Comment by Caspium** [comment]
Here to report the same issue.
Here's my logs: 
[log.txt](https://github.com/user-attachments/files/17084697/log.txt)

Using Sunshine+Moonlight from Windows PC to Moonlight on Google TV
When streaming via moonlight, resolution successfully changes
Upon exiting game, resolution reverts and I'm in Steam Big Picture with my ultrawide resolution

**Comment by rocktale** [comment]
Seems like I have the same issue with Steam Big Picture mode. After playing a game that I started from within Big Picture mode the resolution reverts back from the 4k (on my AppleTV) to the Ultrawide that I usually use from my computer. Closing Big Picture and restarting it (from Moonbeam) solves this issue but it would be nice if the resolution was just kept. What about an option to skip reverting back altogether? My sunshine host is only used remotely without a real monitor attached, thus there is no native / real resolution anyway.

**Comment by Nonary** [comment]
I’m not able to replicate this issue on my machine so I don’t have any idea why people are experiencing a resolution revert before the stream gets terminated because every log that gets posted is showing that it’s not being reverted before they ended the stream

**Comment by xDaveN** [comment]
Not gonna post my log because it's just as inconclusive as the others. But hey, now you know your logging method is somewhat faulty. Where does the script grab the current resolution from?

Seems like the people in here having this issue are all using a non 16:9 native res on their monitors, so maybe that's related.

**Comment by Gman0909** [comment]
I have the same issue as everyone else here, also with a super ultrawide monitor. I have noticed that it only seems to happen in games that use full screen, games that run in borderless window do not cause any issues.

**Comment by KlosiakMK** [comment]
Yeah, only games in exclusive fullscreen mode causes this glitch. I use borderless window too everywhere where I can to avoid this issue.

BTW: The same happens when I use a switch to turn on/off Windows 10 HDR option when stream is ongoing and desktop is streamed.

**Comment by RealZack** [comment]
Hello Nonary,

Same issue, so I enable the debug log in order for you to may find what is going on

Host original resolution 1440p, set the prefered resolution to 1440p in settings.json

Stream steam deck in 1600p, launch a game, exit the game and boom 720p.

I copy the log at this state, even before actually turning of the streaming !

[log_1738095106 - Copy.txt](https://github.com/user-attachments/files/18578279/log_1738095106.-.Copy.txt)

**Comment by Nonary** [comment]
Can you check to make sure the game you're playing isn't using full screen mode? The script cannot work properly if the game you're playing runs in full screen

**Comment by RealZack** [comment]
Certainly, I didn't really check.

I had to reinstall everything because my old setup broke (windows update related ?), i'm just surprise before it was working like a charm without the need to set windows mode ...

**Comment by Nonary** [comment]
I believe this issue is only caused by games that are in exclusive full screen, which is something I cannot realistically fix.
